Assignment6_JUNG.Rmd

```diff
@@ -0,0 +1,131 @@
+
+#If we are worried about overfitting we can remove nodes form our tree using the prune() command, setting cp to the CP value from the table that corresponds to the number of nodes we want the tree to terminate at. Let's set it to two nodes.
+
+```{r}
+c.tree2 <- prune(c.tree1, cp = 0.058182 )#Set cp to the level at which you want the tree to end
+
+#Visualize this tree and compare it to the one you generated earlier
+post(c.tree2, file = "tree2.ps", title = "MOOC") #This creates a pdf image of the tree
+```
+
+#Now use both the original tree and the pruned tree to make predictions about the the students in the second data set. Which tree has a lower error rate?
+
+```{r}
+M2$predict1 <- predict(c.tree1, M2, type = "class")
+
+M2$predict2 <- predict(c.tree2, M2, type = "class")
+
+table(M2$certified, M2$predict1)
+
+table(M2$certified, M2$predict2)
+
+error.rate1 <- 1 - sum(diag(as.matrix(table(M2$certified, M2$predict1)
+))) / sum(table(M2$certified, M2$predict1))
+error.rate1
+
+error.rate2 <- 1 - sum(diag(as.matrix(table(M2$certified, M2$predict2)
+))) / sum(table(M2$certified, M2$predict2))
+error.rate2
+```
+
+Table 2 has lower error rate. 
+
+##Part III
+
+Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
+
+```{r}
+#library
+library(readr)
+
+#Pull data
+MichiganOpenData <- read_csv("https://raw.githubusercontent.com/bkoester/PLA/master/student.course.csv")
+```
+
+From here, I will predict GRD_PTS_PER_UNIT, or the grade received. 
+#For model I, I will use all available variable except AnonID (it would be counterintuitive to predict GPA on the basis on ID, which is designated here as a numerical number.)
+```{r}
+
+c.tree.partIII.1 <- rpart(GRD_PTS_PER_UNIT ~ SUBJECT + CATALOG_NBR + GPAO + DIV + ANON_INSTR_ID + TERM, data = MichiganOpenData)
+```
+
+#print file
+```{r}
+printcp(c.tree.partIII.1)
+```
+
+
+#create pdf image file
+```{r}
+post(c.tree.partIII.1, file = "tree3.ps", title = "Model 1") 
+```
+
+
+Model 2: Since model 1 already ends up only using two variables (GPAO and subject), here, we will only look at GPAO (GPA of other subjects),
+
+```{r}
+
+c.tree.partIII.2 <- rpart(GRD_PTS_PER_UNIT ~ GPAO, data = MichiganOpenData)
+
+#print
+printcp(c.tree.partIII.2)
+
+#post
+post(c.tree.partIII.2, file = "tree4.ps", title = "Model 1") 
+```
+
+Comparing the two trees, the first model is better, with 6 splits and a cv error rate of 0.66648 and a standard error of 0.0014329; the second model has 3 splits, with cv error rate of 0.72934 and a standard error of 0.0014952.
+
```

Comment:
Good work on classification trees! You got the basic idea of model comparison and selection by looking at the error rate and standard error. Next time maybe give reasons for choosing specific features to investigate.